app/services/discourse_translator/provider/amazon.rb

```diff
@@ -120,38 +120,25 @@ def self.detect!(topic_or_post)
         end
       end
 
-      def self.translate_translatable!(translatable, target_locale_sym = I18n.locale)
-        detected_lang = detect(translatable)
+      def self.translate_post!(post, target_locale_sym = I18n.locale, opts = {})
+        raw = opts.key?(:raw) ? opts[:raw] : !opts[:cooked]
```

Comment:
generally if we can get away without this pattern I think it is better..

def self.translate_post!(post, locale: nil, type: :raw)  

something a bit clearer sig wise